src/Type/MixedType.php

```diff
@@ -514,6 +514,10 @@ public function isOffsetAccessible(): TrinaryLogic
 
 	public function hasOffsetValueType(Type $offsetType): TrinaryLogic
 	{
+		if ($this->isOffsetAccessible()->no()) {
```

Comment:
this makes we wonder whether the API is meant in a way that the caller is expected to check with `isOffsetAccessible`  before calling `hasOffsetValueType`